File: tiq-e/array/5/main.ts
import {performance} from 'perf_hooks';

function singleNumber(nums: number[]): number {
  let n = 0
  for (let num of nums) {
    n = n ^ num
  }
  return n
}

let arr = [4, 1, 2, 1, 2];

let start = performance.now();
let result = singleNumber(arr);
let end = performance.now();
console.log('Duration:', end - start);
console.log('Result:', result)

File: tiq-e/array/5/main.go
package main

import (
	"fmt"
	"time"
)

func singleNumber(nums []int) int {
	var out int
	for _, num := range nums {
		out ^= num
	}
	return out
}

func main() {
	arr := []int{3, 2, 2, 6, 3}

	var start = time.Now()
	var result = singleNumber(arr)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
